internal/cluster/cluster.go

```diff
@@ -1399,32 +1403,106 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 	return m.setConnectivityMajorityGroupsForClusterInternal(cluster, db)
 }
 
-func (m *Manager) deleteClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, folder string) error {
-	log := logutil.FromContext(ctx, m.log)
+func (m *Manager) extractManifestFolderAndFilename(ctx context.Context, path string) (string, string, bool) {
+	parts := strings.Split(path, "/")
+	partsLength := len(parts)
+	if partsLength != 4 || parts[1] != "manifests" {
+		return "", "", false
+	}
+	folder := parts[2]
+	filename := parts[3]
+	return folder, filename, true
+}
+
+func (m *Manager) getClusterFilesForFolder(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, folder string) ([]string, error) {
 	path := filepath.Join(string(*c.ID), folder) + "/"
 	files, err := objectHandler.ListObjectsByPrefix(ctx, path)
 	if err != nil {
 		msg := fmt.Sprintf("Failed to list files in %s", path)
 		m.log.WithError(err).Errorf(msg)
-		return common.NewApiError(
+		return nil, common.NewApiError(
 			http.StatusInternalServerError,
 			errors.Errorf(msg))
 	}
+	return files, err
+}
 
-	var failedToDelete []string
+func (m *Manager) ResetClusterFiles(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+	log := logutil.FromContext(ctx, m.log)
+	files, err := m.getClusterFilesForFolder(ctx, c, objectHandler, "")
+	if err != nil {
+		return err
+	}
+	// ListClusterManifestsInternal only returns user generated manifests so we can use this to determine whether or not a manifest should be deleted on reset.
+	userManifests, err := m.manifestApi.ListClusterManifestsInternal(ctx, manifests.V2ListClusterManifestsParams{ClusterID: *c.Cluster.ID})
```

Comment:
It seems like it would be easier to create a manifestsApi function to return the system manifests then just delete them directly.

This strategy of getting the user manifests then iterating through all the files eliminating those is strange. Is there any other reason you're doing it this way?

Reply:
What about an API function to delete system manifests directly via API, maybe that would work better?